code/game/objects/items/weapons/power_cells.dm

```diff
@@ -211,3 +204,48 @@
 	. = ..()
 	charge = 0
 	update_icon()
+
+/obj/item/cell/mecha
+	name = "power core"
+	origin_tech = list(TECH_POWER = 2)
+	icon_state = "core"
+	w_class = ITEMSIZE_LARGE
+	maxcharge = 20000
+	matter = list(DEFAULT_WALL_MATERIAL = 20000, MATERIAL_GLASS = 10000)
+
+	/// When the cell will next recharge (once every minute)
+	var/next_recharge
+
+	/// Percentage of maxcharge to recharge every minute, leave null for no self-recharge
+	var/self_charge_percentage
+
+/obj/item/cell/mecha/Initialize()
+	. = ..()
+	if(self_charge_percentage)
+		START_PROCESSING(SSprocessing, src)
+
+/obj/item/cell/mecha/Destroy()
+	if(self_charge_percentage)
+		STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/obj/item/cell/mecha/process()
+	if(self_charge_percentage && next_recharge < world.time)
+		give((maxcharge / 100) *self_charge_percentage)
+		next_recharge = world.time + 1 MINUTE
```

Comment:
I'm not sure there's much point in having SSprocess call this over and over only to wait for a predetermined time (which is wasted overhead), instead of using a timer that only calls it when it actually needs to process

If you want to use process this way, it must respect what the DMdoc of it says:

```
* Since the wait setting of a subsystem can be changed at any time, it is important that any rate-of-change that you implement in this proc is multiplied by the seconds_per_tick that is sent as a parameter,
 * Additionally, any "prob" you use in this proc should instead use the SPT_PROB define to make sure that the final probability per second stays the same even if the subsystem's wait is altered.
 * Examples where this must be considered:
 * - Implementing a cooldown timer, use `mytimer -= seconds_per_tick`, not `mytimer -= 1`. This way, `mytimer` will always have the unit of seconds
 * - Damaging a mob, do `L.adjustFireLoss(20 * seconds_per_tick)`, not `L.adjustFireLoss(20)`. This way, the damage per second stays constant even if the wait of the subsystem is changed
 * - Probability of something happening, do `if(SPT_PROB(25, seconds_per_tick))`, not `if(prob(25))`. This way, if the subsystem wait is e.g. lowered, there won't be a higher chance of this event happening per second
 *
```
So nuke the `next_recharge `out, get the `seconds_per_tick` parameter, and multiply the `self_charge_percentage `by it, this will maintain the recharge rate consistent with real time